src/coreclr/jit/compiler.cpp

```diff
@@ -5630,11 +5630,8 @@ int Compiler::compCompile(CORINFO_MODULE_HANDLE classPtr,
 
     // Set this before the first 'BADCODE'
     // Skip verification where possible
-    //.tiVerificationNeeded = !compileFlags->IsSet(JitFlags::JIT_FLAG_SKIP_VERIFICATION);
     assert(compileFlags->IsSet(JitFlags::JIT_FLAG_SKIP_VERIFICATION));
 
-    assert(!compIsForInlining() || !tiVerificationNeeded); // Inlinees must have been verified.
```

Comment:
Since `tiVerificationNeeded` has basically been `false` for a long time, we assume `!tiVerificationNeeded` will always be `true`.

There are a few asserts that have an 'or' condition of `!tiVerificationNeeded` which would mean that the assert has always been passing. In order to maintain the current behavior, we remove the assert altogether.